packages/back-end/src/jobs/updateScheduledFeatures.ts

```diff
@@ -0,0 +1,102 @@
+import Agenda, { Job } from "agenda";
+import { FeatureModel } from "../models/FeatureModel";
+import {
+  featureUpdated,
+  getEnabledEnvironments,
+  getNextScheduledUpdate,
+} from "../services/features";
+import { logger } from "../util/logger";
+
+type UpdateSingleFeatureJob = Job<{
+  featureId: string;
+}>;
+
+const QUEUE_FEATURE_UPDATES = "queueScheduledFeatureUpdates";
+
+const UPDATE_SINGLE_FEATURE = "updateSingleFeature";
+
+async function fireUpdateWebhook(agenda: Agenda) {
+  const updateFeatureJob = agenda.create(QUEUE_FEATURE_UPDATES, {});
+  updateFeatureJob.unique({});
+  updateFeatureJob.repeatEvery("1 minute");
+  await updateFeatureJob.save();
+}
+
+async function queueFeatureUpdate(agenda: Agenda, featureId: string) {
+  const job = agenda.create(UPDATE_SINGLE_FEATURE, {
+    featureId,
+  }) as UpdateSingleFeatureJob;
+
+  job.unique({
+    featureId,
+  });
+  job.schedule(new Date());
+  await job.save();
+}
+
+export default async function (agenda: Agenda) {
+  agenda.define(QUEUE_FEATURE_UPDATES, async () => {
+    const featureIds = (
+      await FeatureModel.find({
```

Comment:
Not a big deal, but it does feel like .find() should be happening from within FeatureModel.ts